src/handlers/backport.rs

```diff
@@ -0,0 +1,227 @@
+use crate::handlers::Context;
+use regex::Regex;
+use tracing as log;
+
+// see https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue
+static CLOSES_ISSUE_REGEXP: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new("(?i)(close[sd]*|fix([e]*[sd]*)?|resolve[sd]*) #(\\d+)").unwrap());
+
+const BACKPORT_LABELS: [&str; 4] = [
+    "beta-nominated",
+    "beta-accepted",
+    "stable-nominated",
+    "stable-accepted",
+];
+
+#[derive(Default)]
+pub(crate) struct BackportInput {
+    // Issue(s) fixed by this PR
+    ids: Vec<u64>,
+    // Labels profile, compound value of (needs_label -> add_labels)
+    profile_labels: HashMap<String, Vec<String>>,
+}
+
+pub(super) async fn parse_input(
+    _ctx: &Context,
+    event: &IssuesEvent,
+    config: Option<&BackportTeamConfig>,
+) -> Result<Option<BackportInput>, String> {
+    let config = match config {
+        Some(config) => config,
+        None => return Ok(None),
+    };
+
+    if !matches!(event.action, IssuesAction::Opened) && !event.issue.is_pr() {
+        log::info!(
+            "Skipping backport event because: IssuesAction = {:?} issue.is_pr() {}",
+            event.action,
+            event.issue.is_pr()
+        );
+        return Ok(None);
+    }
+    let pr = &event.issue;
+
+    let pr_labels: Vec<&str> = pr.labels.iter().map(|l| l.name.as_str()).collect();
+    if contains_any(&pr_labels, &BACKPORT_LABELS) {
+        log::debug!("PR #{} already has a backport label", pr.number);
+        return Ok(None);
+    }
+
+    // Retrieve backport config for this PR, based on its team label(s)
+    // If the PR has no team label matching any [backport.*.team_labels] config, the backport labelling will be skipped
+    let mut input = BackportInput::default();
+    let valid_configs: Vec<_> = config
+        .configs
+        .iter()
+        .clone()
+        .filter(|(_cfg_name, cfg)| {
+            let team_labels: Vec<&str> = cfg.team_labels.iter().map(|l| l.as_str()).collect();
+            if !contains_any(&pr_labels, &team_labels) {
+                log::warn!(
+                    "Skipping backport nomination: PR is missing one required team label: {:?}",
+                    pr_labels
+                );
+                return false;
+            }
+            input
+                .profile_labels
+                .insert(cfg.needs_label.clone(), cfg.add_labels.clone());
+            true
+        })
+        .collect();
+    if valid_configs.is_empty() {
+        log::warn!(
+            "Skipping backport nomination: could not find a suitable backport config. Please ensure the triagebot.toml has a `[backport.*.team_labels]` section matching the team label(s) for PR #{}.",
+            pr.number
+        );
+        return Ok(None);
+    }
+
+    // Check marker text in the opening comment of the PR to retrieve the issue(s) being fixed
+    for caps in CLOSES_ISSUE_REGEXP.captures_iter(&event.issue.body) {
```

Comment:
FWIW, capturing over the issue body without taking into account the markdown is insufficient as people might use `Fixes #123` and not expect us to trigger on it.

Preventing that is bit finicky so let's defer that.

Reply:
For reference: do you have an example? So I can look at that later.

Reply:
oh wait maybe you mean people actually writing in the comment:
```
`Fixes #123`
```
to avoid the GitHub automated issue cross-reference?